src/LendingPool.sol

```diff
@@ -172,13 +184,14 @@ contract LiquidityPool is ERC20, Owned {
 
         asset.safeTransferFrom(from, address(this), assets);
 
-        _mint(msg.sender, assets);
+        supplyBalances[msg.sender] += assets;
+        totalSupply += assets;
 
         _updateInterestRate();
     }
 
     /**
-     * @notice Withdraw assets from the Liquidity Pool
+     * @notice Withdraw assets from the Lending Pool
```

Comment:
Since this function is not according to the ERC20 standard anymore, we can remove owner_ and only work with msg.sender